File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        unordered_map<string, vector<string> > mp;
        for(string &str : strs){
            vector<int> t(26,0);
            for(char ch: str){
                t[ch - 'a']++;
            }
            string key = "";
            for(int i : t){
                key += "#" + to_string(i);
            }
            mp[key].push_back(str);
        }
        for(auto it: mp){
            ans.push_back(it.second);
        }
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String> > mp = new HashMap<>();
        List<List<String>> ans = new ArrayList<>();
        for(int i = 0; i < strs.length; i++){
            char[] chars = strs[i].toCharArray();
            Arrays.sort(chars);
            String s = new String(chars);
            if(mp.containsKey(s)){
                mp.get(s).add(strs[i]);

            }else{
                ArrayList<String> t = new ArrayList<>();
                t.add(strs[i]);
                mp.put(s, t);
            }
        }

        for(List<String> val : mp.values()){
            ans.add(val);
        }
        return ans;
    }
}